public/app/features/dashboard-scene/settings/JsonModelEditView.tsx

```diff
@@ -0,0 +1,76 @@
+import { css } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2, PageLayoutType } from '@grafana/data';
+import { SceneComponentProps, SceneObjectBase } from '@grafana/scenes';
+import { Button, CodeEditor, useStyles2 } from '@grafana/ui';
+import { Page } from 'app/core/components/Page/Page';
+import { Trans } from 'app/core/internationalization';
+import { getPrettyJSON } from 'app/features/inspector/utils/utils';
+
+import { DashboardScene } from '../scene/DashboardScene';
+import { transformSceneToSaveModel } from '../serialization/transformSceneToSaveModel';
+import { getDashboardSceneFor } from '../utils/utils';
+
+import { DashboardEditView, DashboardEditViewState, useDashboardEditPageNav } from './utils';
+
+export interface JsonModelEditViewState extends DashboardEditViewState {}
+
+export class JsonModelEditView extends SceneObjectBase<JsonModelEditViewState> implements DashboardEditView {
+  public getUrlKey(): string {
+    return 'json-model';
+  }
+
+  public getDashboard(): DashboardScene {
+    return getDashboardSceneFor(this);
+  }
+
+  static Component = ({ model }: SceneComponentProps<JsonModelEditView>) => {
+    const dashboard = model.getDashboard();
+    const jsonData = transformSceneToSaveModel(dashboard);
+    const jsonText = getPrettyJSON(jsonData);
```

Comment:
Nit - can be extracted to a model method, `model.getJsonText()`, to make the renderer as stupid as possible?

Reply:
sure, i'll fix that